src/MergeContext.js

```diff
@@ -501,13 +501,15 @@ class PullRequest {
     async _checkApproval() {
         assert(this._approval === null);
 
-        const collaborators = await GH.getCollaborators();
-        const pushCollaborators = collaborators.filter(c => c.permissions.push === true);
         const requestedReviewers = this._prRequestedReviewers();
 
-        for (let collaborator of pushCollaborators) {
-            if (requestedReviewers.includes(collaborator.login)) {
-                this._log("requested core reviewer: " + collaborator.login);
+        for (let reviewerPair of requestedReviewers.entries()) {
+            const reviewer = reviewerPair[0];
+            const requestedReviewerIsCore = Config.coreDeveloperIds().get(reviewer);
+            if (requestedReviewerIsCore) {
+                const reviewerId = reviewerPair[1];
+                assert(requestedReviewerIsCore === reviewerId);
+                this._log("requested core reviewer: " + reviewer);
```

Comment:
Please move this debugging up, _before_ the assert(). If we can print reviewerPair instead (and get both values), we should.

Reply:
I removed this line because the new _coreDeveloper() does this debugging.